src/Pos/Explorer/Web/Server.hs

```diff
@@ -250,6 +304,45 @@ getTxSummary cTxId = do
 -- Helpers
 --------------------------------------------------------------------------------
 
+-- | Find all `MainBlock` by applying the *predicate*, starting from *headerHash*
+filterMainBlocks :: (SscHelpersClass ssc, ExplorerMode m) =>
+    HeaderHash ->
+    (MainBlock ssc -> Bool) ->
+    m [MainBlock ssc]
+filterMainBlocks headerHash predicate = rights <$> generalBlockSearch
+  where
+    generalBlockSearch = filterAllBlocks headerHash specializedPred (pure [])
+    specializedPred block = case block of
+        Left  _   -> False
+        Right mb  -> predicate mb
+
+-- | Find all blocks matching the sent predicate. This is a generic function
+-- that can be called with either `MainBlock` or `GenesisBlock` in mind.
+filterAllBlocks :: (SscHelpersClass ssc, ExplorerMode m) =>
+    HeaderHash ->
+    (Block ssc -> Bool) ->
+    m [Block ssc] ->
+    m [Block ssc]
+filterAllBlocks headerHash predicate acc
+    -- When we reach the genesis block, return the accumulator.
```

Comment:
Actually I meant, just write it here, but code comment is even better.
Well, I thought you discovered in #explorer that genesis block is present in each *epoch* once, not in blockchain once, or I'm wrong?
https://input-output-rnd.slack.com/archives/C3NJV7ADQ/p1490368930079121